enumerable_custom.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  def my_each
+    array = to_a
+    if block_given?
+      array.each { |i| yield(i) }
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    array = to_a
+    i = 0
+    if block_given?
+      until i == array.length
+        yield(array[i], i)
+        i += 1
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select(*)
+    if block_given?
+      array = []
+      my_each { |item| array << item if yield(item) }
+    else
+      to_enum(:my_select)
+    end
+    array
+  end
+
+  def my_all?
+    if block_given?
+      condition = true
+      my_each { |item| condition = false if yield(item) == false }
+    else
+      to_enum(:my_all?)
+    end
+    condition
+  end
+
+  def my_any?
+    if block_given?
+      condition = false
+      my_each { |item| condition = true if yield(item) == true }
+    else
+      to_enum(:my_any?)
+    end
+    condition
+  end
+
+  def my_none?
+    condition = true
+    my_each { |item| condition = false if yield(item) }
+    condition
+  end
```

Comment:
The issues with **my_none?**, **my_any?** and **my_all?** are similar. the methods should be able to: 

* [ ] Return a suitable boolean result if no arguments were passed.
  ```ruby
	array = [false, true, false, nil, []]
	puts array.my_any? # should print true
	puts array.my_all? # should print false
	puts array.my_none? # should print false
  ```
* [ ] Accept primitives, classes (i.e Integer, String, etc...) and Regex Expressions as arguments.
  ```ruby
	words = %w[dog door rod blade]
	puts words.my_all?(/c/) == words.all?(/c/) # false
	puts words.my_any?(/c/) == words.all?(/c/) # false
	puts words.my_none?(/c/) == words.all?(/c/) # true
  ```